### PR TITLE
Remove strongly type documents support from indexDocuments related operations

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: sdk/search/azure-search-documents/README.md [hunk 1]
```diff
@@ -153,21 +153,27 @@ searchIndexClient.createIndex(newIndex);
...
+<!-- embedme ./src/samples/java/com/azure/search/documents/ReadmeSamples.java#L131-L142 -->
 ```java
-List<Hotel> hotels = new ArrayList<>();
-hotels.add(new Hotel().setHotelId("100"));
-hotels.add(new Hotel().setHotelId("200"));
-hotels.add(new Hotel().setHotelId("300"));
+List<SearchDocument> documents = new ArrayList<>();
+documents.add(new SearchDocument(new HashMap<String, String>(){{
+    put("HotelId", "100");
+}}));
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/README.md:161]: Collections.singletonMap may be a better option here.

```suggestion
documents.add(new SearchDocument(Collections.singletonMap("HotelId", "100")));
```

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedFlux.java [hunk 2]
```diff
@@ -21,7 +20,7 @@
  * element type is {@link SearchResult}, and the page type is {@link SearchPagedResponse}.
  */
 public final class SearchPagedFlux extends PagedFluxBase<SearchResult, SearchPagedResponse> {
-    private final Supplier<Mono<SearchFirstPageResponseWrapper>> metadataSupplier;
+    private final Supplier<Mono<SearchPagedResponse>> metadataSupplier;
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedFlux.java:23]: Did you mean to make this change?

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedFlux.java:23]: If so should `SearchFirstPageResponseWrapper` be deleted?

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedFlux.java:23]: No, we still need the wrapper. We cached the first paged instance (in wrapper class) at SearchAsyncClient search method. The first page retriever has set the right first paged instance from top-level. So we don't need to set it here anymore.

**Pull request closed without merging**